### ISSUE TITLE
Evaluation error when reimplementing kitti

### ISSUE THREAD

**Issue by xyaochen** [open-mmlab/mmdetection3d#1227]
**Describe the issue**
The results are confusing when I run models on kitti dataset. Either I use my own models and my own configs or use mmdet3d provided configs, like SECOND, the val results are not converging.

**Reproduction**
1. What command or script did you run?

bash tools/dist_train.sh configs/second/hv_second_secfpn_6x8_80e_kitti-3d-3class.py 8

2. What config dir you run?

https://github.com/open-mmlab/mmdetection3d/blob/master/configs/second/hv_second_secfpn_6x8_80e_kitti-3d-3class.py

3. Did you make any modifications on the code or config? Did you understand what you have modified?
No
4. What dataset did you use?
kitti
**Environment**

sys.platform: linux
Python: 3.6.9 (default, Oct  8 2020, 12:12:24) [GCC 8.4.0]
CUDA available: True
GPU 0,1,2,3,4,5,6,7: NVIDIA A100-PCIE-40GB
CUDA_HOME: /usr/local/cuda
NVCC: Build cuda_11.1.TC455_06.29069683_0
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.8.0a0+17f8c32
PyTorch compiling details: PyTorch built with:
  - GCC 7.5
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.1 Product Build 20180928 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash N/A)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_52,code=sm_52;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_86,code=compute_86
  - CuDNN 8.0.4
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, FORCE_FALLBACK_CUDA_MPI=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=ON, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.0a0
OpenCV: 4.5.2
MMCV: 1.3.12
MMCV Compiler: GCC 7.5
MMCV CUDA Compiler: 11.1
MMDetection: 2.14.0
MMSegmentation: 0.14.1
MMDetection3D: 0.17.1+45e7007

**Results**
the val results by SECOND on kitti are as follows:
![image](https://user-images.githubusercontent.com/73167393/153249942-2ee1b304-7de1-4f73-891d-87470597ead8.png)
while the training loss are in accordance with the released log file

![image](https://user-images.githubusercontent.com/73167393/153250290-d767090b-d606-4ae1-9192-ef1b6e9ce2ce.png)
https://download.openmmlab.com/mmdetection3d/v0.1.0_models/second/hv_second_secfpn_6x8_80e_kitti-3d-3class/hv_second_secfpn_6x8_80e_kitti-3d-3class_20200620_230238.log.json

The evaluation results by my own models have similar problems.

If the evaluation function of kitti dataset or the metric computation have been modified while the config or model are not revised in time?

**Comment by ZCMax** [comment]
Sorry for the late reply, can you please provide your training log file?

**Comment by xyaochen** [comment]
Sure, the log file and config are here
[second_log.zip](https://github.com/open-mmlab/mmdetection3d/files/8048594/second_log.zip)
Because I'm running in a computer cluster, the dataset is stored as big files. I'm sure that the data loading is not the reason.

**Comment by xyaochen** [comment]
> Sorry for the late reply, can you please provide your training log file?

Sorry to bother you, could you take a look? I'm still confused for the reimplementation on kitti

**Comment by Tai-Wang** [comment]
It's really strange. Could you please visualize the results and see what happens? Do you follow our documentation to generate data and infos and always use mmdet3d 0.17.1?

**Comment by Tai-Wang** [comment]
Close due to inactivity.
